index.js

```diff
@@ -36,17 +38,34 @@ const parseArgs = (
       arg = arg.replace(/^-+/, '')
 
       if (arg.includes('=')) {
+        //withValue equals(=) case
         const argParts = arg.split('=')
 
         result.args[argParts[0]] = true
-        if (options.withValue) {
-          result.values[argParts[0]] = argParts[1]
-        }
-      }
-      else {
-        result.args[arg] = true
+        //If withValue option is specified, take 2nd part after '=' as value, else set value as undefined
+        const val = options.withValue && options.withValue.includes(argParts[0]) ? argParts[1] : undefined
+        //Append value to previous arg values array for case of multiples option, else add to empty array
+        result.values[argParts[0]] = [...(options.multiples && options.multiples.includes(argParts[0]) && result.values[argParts[0]] ? result.values[argParts[0]] : []), val]
```

Comment:
this is correct, but this is a common JS idiom that will be faster by not resorting to the iterator protocol:
```suggestion
        result.values[argParts[0]] = [].concat(
          options.multiples && options.multiples.includes(argParts[0]) && result.values[argParts[0]] || [],
          val,
        )
```